packages/grafana-ui/src/components/Graph/GraphLegend.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useContext } from 'react';
+import { LegendProps, LegendItem, LegendDisplayMode } from '../Legend/Legend';
+import { GraphLegendListItem, GraphLegendTableRow } from './GraphLegendItem';
+import { SeriesColorChangeHandler, SeriesAxisToggleHandler } from './GraphWithLegend';
+import { LegendTable } from '../Legend/LegendTable';
+import { LegendList } from '../Legend/LegendList';
+import union from 'lodash/union';
+import sortBy from 'lodash/sortBy';
+import { ThemeContext } from '../../themes/ThemeContext';
+import { css } from 'emotion';
+import { selectThemeVariant } from '../../themes/index';
+
+interface GraphLegendProps extends LegendProps {
+  displayMode: LegendDisplayMode;
+  sortBy?: string;
+  sortDesc?: boolean;
+  onSeriesColorChange: SeriesColorChangeHandler;
+  onSeriesAxisToggle?: SeriesAxisToggleHandler;
+  onToggleSort: (sortBy: string) => void;
+  onLabelClick: (item: LegendItem, event: React.MouseEvent<HTMLElement>) => void;
+}
+
+export const GraphLegend: React.FunctionComponent<GraphLegendProps> = ({
+  items,
+  displayMode,
+  sortBy: sortKey,
+  sortDesc,
+  onToggleSort,
+  onSeriesAxisToggle,
+  placement,
+  className,
+  ...graphLegendItemProps
+}) => {
+  const theme = useContext(ThemeContext);
+
+  if (displayMode === LegendDisplayMode.Table) {
+    const columns = items
+      .map(item => {
+        if (item.displayValues) {
+          return item.displayValues.map(i => i.title);
+        }
+        return [];
+      })
+      .reduce(
+        (acc, current) => {
+          return union(acc, current.filter(item => !!item));
+        },
+        ['']
+      ) as string[];
+
+    const sortedItems = sortKey
+      ? sortBy(items, item => {
+          if (item.displayValues) {
+            const stat = item.displayValues.filter(stat => stat.title === sortKey)[0];
+            return stat && stat.numeric;
+          }
+          return undefined;
+        })
+      : items;
```

Comment:
if `sortKey` was an index, i think it would simplify trying to lookup by a title.  Would also fix any issues with duplicate or missing titles :)

Reply:
Will fix that in separate PR